lib/features/judge/data/local/judge_repository_imp.dart

```diff
@@ -28,9 +35,16 @@ class JudgeRepositoryImp extends JudgeRepository {
       });
       final DocumentSnapshot result = await docRef.get();
 
-      if (result.data() != null) {
+      //Upload image to firebase storage
+      taskSnapshot = await imageRef.putFile(image);
+
+      if (result.data() != null && taskSnapshot.state == TaskState.success) {
         var id = result.id;
-        await db.collection(judges).doc(id).update({"id": id});
+        var urlPhoto = await imageRef.getDownloadURL();
+        await db
+            .collection(judges)
+            .doc(id)
+            .update({"id": id, "urlPhoto": urlPhoto});
```

Comment:
Excellent just take in mind that in the future other view could use something similar, so you want to probably create a repository to do this "upload a file and return a string"

Reply:
I tried to do it, please tell me if it is the right way to do it